### ISSUE TITLE
Defective Schedule error repeatedly

### ISSUE THREAD

**Issue by tmchow** [nielsfaber/scheduler-component#102]
Since installing the latest beta (3.1.0b), I cannot create a schedule for a thermostat heater that used to work before this upgrade.  I've tried deleting and recreating but same issue repeatedly.

### Does it happen with specific entities and/or actions or for all created rules?
Seems to be only with specific entities.

### Affects only time schemes or also simple schedules? Sunrise involved? Do you use conditions?
I haven't done extensive testing but my other scheduling works.  It's happening for "schemes" only for me but other schemes are working so it's not schemes that are broken entirely.

### Does it matter which days did you set up? Are you using the workday integration?
I'm using the workday integration, and one thing that's consistent is that I tried diong a new schedule for a working entity in another schedule but it fails when I try Scheme + `Weekend`.

### Could you share screenshots from Developer Tools → Entities showing the properties of the schedules that are marked defective by the card?

![image](https://user-images.githubusercontent.com/517103/114593422-f2c88680-9c40-11eb-97a5-1eadf0967f58.png)

![image](https://user-images.githubusercontent.com/517103/114593477-ffe57580-9c40-11eb-9450-2f6008a99a7e.png)

### Any clues in the HA logs?

I see this in the logs:

```
2021-04-12 12:25:51 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new switch.scheduler entity: switch.schedule_kids_heater_weekend
2021-04-12 12:25:51 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/config/custom_components/scheduler/timer.py", line 90, in async_reload_data
await self.async_start_timer()
File "/config/custom_components/scheduler/timer.py", line 100, in async_start_timer
[current_slot, timestamp_end] = self.current_timeslot()
File "/config/custom_components/scheduler/timer.py", line 378, in current_timeslot
self.calculate_timestamp(slot[const.ATTR_STOP], now)
File "/config/custom_components/scheduler/timer.py", line 334, in calculate_timestamp
return self.calculate_timestamp(time_str, next_day)
File "/config/custom_components/scheduler/timer.py", line 334, in calculate_timestamp
return self.calculate_timestamp(time_str, next_day)
File "/config/custom_components/scheduler/timer.py", line 334, in calculate_timestamp
return self.calculate_timestamp(time_str, next_day)
[Previous line repeated 974 more times]
File "/config/custom_components/scheduler/timer.py", line 288, in calculate_timestamp
ts = dt_util.find_next_time_expression_time(
File "/usr/src/homeassistant/homeassistant/util/dt.py", line 325, in find_next_time_expression_time
result = tzinfo.localize(result, is_dst=None)
File "/usr/local/lib/python3.8/site-packages/pytz/tzinfo.py", line 328, in localize
loc_dt = tzinfo.normalize(dt.replace(tzinfo=tzinfo))
File "/usr/local/lib/python3.8/site-packages/pytz/tzinfo.py", line 256, in normalize
return self.fromutc(dt)
File "/usr/local/lib/python3.8/site-packages/pytz/tzinfo.py", line 199, in fromutc
idx = max(0, bisect_right(self._utc_transition_times, dt) - 1)
RecursionError: maximum recursion depth exceeded in comparison
```

**Comment by nielsfaber** [comment]
Thanks for this!
I'm quite sure I have all info to reproduce and tackle the issue.
You can always roll back to the previous version for the time being (this one is a beta), I'm hoping to fix this issue in the weekend.

**Comment by nielsfaber** [comment]
I could reproduce this problem locally and added a fix in https://github.com/nielsfaber/scheduler-component/commit/e370ec3bb83e0a08177966a8d8e7a3d186d857f5.

**Comment by nielsfaber** [comment]
I just released v3.1.0 which contains this fix.

**Comment by nielsfaber** [comment]
I will close this issue, assuming the problem is fixed.
Let me know if you still see this problem.
